app/controllers/public.js

```diff
@@ -1,24 +1,28 @@
 import Controller from '@ember/controller';
 import moment from 'moment';
-import { computed } from '@ember/object';
+import { action, computed } from '@ember/object';
 
-export default Controller.extend({
-  smallLead: computed('session.currentRouteName', function() {
-    if (this.get('session.currentRouteName')) {
-      return this.get('session.currentRouteName') !== 'public.index';
+export default class extends Controller {
+  @computed('session.currentRouteName')
+  get smallLead() {
+    if (this.session.currentRouteName) {
+      return this.session.currentRouteName !== 'public.index';
     }
-  }),
-  displayEndDate: computed('model.startsAtDate', 'model.endsAtDate', function() {
+    return null;
+  }
+  @computed('model.startsAtDate', 'model.endsAtDate')
+  get displayEndDate() {
     return !moment(this.model.startsAtDate).isSame(this.model.endsAtDate, 'minute');
-  }),
-  displaySideMenu: computed('session.currentRouteName', function() {
+  }
+  @computed('session.currentRouteName')
```

Comment:
Observe all the properties which are getting used in the computed function.

Reply:
@kushthedude please help me with observers . I am not getting any reference that how to implement them here in ES6 .
the below one that i tried with js.
```
displayEndDateChanges: observer('model.startsAtDate', 'model.endsAtDate', function() {
    console.log(this.model.startsAtDate + " " + this.model.endsAtDate)
  }),
```

Reply:
Didn't quite get you here ?

Reply:
> Observe all the properties which are getting used in the computed function.

@kushthedude what is meaning of `observe all properties` .If it is related to observers , how to implement that here ?